### PR TITLE
[FEATURE] Support underscore in the `condition_value` of a `row_condition`

### PR DESCRIPTION
**Pull request by sp1thas**
Changes proposed in this pull request:
- Allow underscore charachter (`_`) in a condition_value

This PR aims to allow the user to add an underscore in the condition value of a row condition. For instance, after merging this PR, the following expectation is going to work instead of raising: `ConditionParserError: unable to parse condition: col("foo")=="a_b"`

```json
{
  "data_asset_type": null,
  "expectation_suite_name": "foo",
  "expectations": [
    {
      "expectation_type": "expect_column_values_to_be_between",
      "kwargs": {
        "column": "foo",
        "min_value": 1,
        "max_value": 2,
        "condition_parser": "great_expectations__experimental__",
        "row_condition": "col(\"foo\")==\"a_b\""
      }
    }
  ]
}
```

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [ ] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/expectations/row_conditions.py
Comment (AFineDayFor) [inline great_expectations/expectations/row_conditions.py:45]: LGTM 👍

**Review by AFineDayFor — APPROVED** [review]
Thank you very much for identifying this and addressing it :bow:

**Comment by andrewm4894** [comment]
What would be involved in getting this merged? 

Maybe @sp1thas would need to rebase to develop?

I have some long format tables and being able to allow "_" where each row is a metric name of some sort will be really useful.

**Comment by sp1thas** [comment]
I was about to ask the same @andrewm4894 . Based on #5387 (which is quite recent), merging commis into `develop` from `<GITHUB-USER>:<FEATURE-BRANCH>` seems to be the case. I guess another maintainer should review this PR too. @AFineDayFor do you have any clue?

**Pull request merged**
